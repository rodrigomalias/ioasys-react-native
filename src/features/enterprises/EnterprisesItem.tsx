import { useNavigation } from "@react-navigation/native"
import { colors } from "../../colors"
import { IEnterprise } from "models/enterprise/Ienterprise"
import * as React from "react"
import { Image, StyleSheet, Text, TouchableOpacity, View } from "react-native"
import { Icon } from "react-native-elements"

interface IEnterprisesItem {
    enterprise: IEnterprise
}

export const EnterprisesItem = (props: IEnterprisesItem) => {
    const { enterprise } = props
    const navigation = useNavigation()

    const imageUrl = `https://empresas.ioasys.com.br/${enterprise.photo}`

    const handleLogin = React.useCallback(() => {
        navigation.navigate("enterprise", {
            enterpriseName: enterprise.enterprise_name,
            enterpriseId: enterprise.id,
        })
    }, [ enterprise.enterprise_name, enterprise.id, navigation ])

    return (
        <TouchableOpacity onPress={handleLogin} style={styles.enterpriseContainer}>
            <View style={styles.imageContainer}>
                <Image source={{ uri: imageUrl }} style={styles.image} />
            </View>
            <View style={styles.nameContainer}>
                <Text style={styles.name}>
                    {enterprise.enterprise_name}
                </Text>
                <Text style={styles.enterpriseType}>
                    {enterprise.enterprise_type.enterprise_type_name}
                </Text>
            </View>
            <View style={ styles.columnArrow }>
                <Icon
                    name='angle-right'
                    type='font-awesome'
                    size={40}
                    color={colors.darkGray}/>
            </View>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    enterpriseContainer: {
        flex: 1,
        margin: 5,
        height: 80,
        flexDirection: "row",
        borderBottomWidth: 1,
        backgroundColor: colors.white,
        borderBottomColor: colors.lightGray,
        borderRadius: 10,
        shadowColor: colors.black,
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5,
    },
    imageContainer: {
        flex: 1,
        alignItems: "center",
        justifyContent: "center",
    },
    nameContainer: {
        flex: 1.5,
        justifyContent: "center",
    },
    name: {
        fontSize: 16,
        fontWeight: "bold",
    },
    enterpriseType: {
        fontSize: 14,
    },
    image: {
        width: 60,
        height: 60,
        borderRadius: 50,
        borderWidth: 1,
        borderColor: colors.lightGray,
        shadowColor: colors.black,
        shadowOffset: {
            width: 2,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
    },
    columnArrow: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
    }
})