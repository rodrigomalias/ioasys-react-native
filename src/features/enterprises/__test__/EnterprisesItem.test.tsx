import React from "react"
import { EnterprisesItem } from "../EnterprisesItem"
import { render } from "@testing-library/react-native"
import renderer from "react-test-renderer"

const mockValue = {
    city: "SP",
    country: "Brazil",
    description: "Description test",
    email_enterprise: "email@email.com",
    enterprise_name: "Company test",
    enterprise_type: {
        enterprise_type_name: "Type test",
        id: 1
    },
    facebook: undefined,
    id: 1,
    linkedin: undefined,
    own_enterprise: false,
    phone: undefined,
    photo: "photo.png",
    share_price: 5000,
    shares: undefined,
    twitter: undefined,
    value: 500
}

describe("<EnterprisesItem/>", () => {
    test("should render correctly", () => {
        const tree = renderer.create(<EnterprisesItem enterprise={mockValue}/>).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test("should find the image using testId", () => {
        const { getByTestId } = render(<EnterprisesItem enterprise={mockValue} />)

        const image = getByTestId("test_enterprises_item_image")

        expect(image).not.toBeNull()
    })

    test("should find the button using testId", () => {
        const { getByTestId } = render(<EnterprisesItem enterprise={mockValue} />)

        const image = getByTestId("test_enterprises_item_button")

        expect(image).not.toBeNull()
    })
})