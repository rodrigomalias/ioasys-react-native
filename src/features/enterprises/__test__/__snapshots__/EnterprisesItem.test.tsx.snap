// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnterprisesItem/> should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#C0C0C0",
      "borderBottomWidth": 1,
      "borderRadius": 10,
      "elevation": 5,
      "flex": 1,
      "flexDirection": "row",
      "height": 80,
      "margin": 5,
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
  testID="test_enterprises_item_button"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://empresas.ioasys.com.br/photo.png",
        }
      }
      style={
        Object {
          "borderColor": "#C0C0C0",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 60,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "width": 60,
        }
      }
      testID="test_enterprises_item_image"
    />
  </View>
  <View
    style={
      Object {
        "flex": 1.5,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Company test
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      Type test
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#999",
                  "fontSize": 40,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            ï„…
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
